### PR TITLE
(feat) service-example

### PR DESCRIPTION
**Pull request by radotzki**

### CODE REVIEW

File: client/app/common/user/user.factory.js [hunk 1]
```diff
@@ -1,15 +1,19 @@
 let UserFactory = function () {
```
Comment (fesor) [inline client/app/common/user/user.factory.js]: I think that we should use service here instead of factory.

**Comment by radotzki** [comment]
> "There is no User service available in about module. Please specify dependencies."

I want to show that you can use a "global" service in your application (which was included in the main app.js)

> "service name is represents an instance of object, not object constructor. This should be aboutService instead."

I agree, will change.

> "why this service should be stateful?"

The alternative will be:

```javascript
const aboutService = function(userSerivce) {
    'ngInject';
    this.user = userSerivce;
    this.name = this.user.getUserName();

    this.getName = getName;

    function getName() {
        return `About: ${this.name}`;
    }
};
```

My approach is similar to angular2 syntax (where service is just a simple class).
What do you think about it?


> "I think that we should use service here instead of factory."

I agree, will change.

**Comment by fesor** [comment]
> I want to show that you can use a "global" service in your application

Not sure is it a bad practice or just normal case. I have such things in my projects sometime and I don't like it. With this approach we lose dependencies between different modules of application. This could be OK for application layer, but we are trying to make reusable components so they should be independent. I think this need third point of view.

> The alternative will be:

By stateless I mean that your service example has it's own state:

```
this.name = this.user.getUsername();
```

**Comment by radotzki** [comment]
How can you inject service inside another service and stay stateless?

**Comment by fesor** [comment]
@radotzki state means it could be changed in time. If you inject some service you don't have state. But if you get some data from service and store it in another, you have state and now you should track data for updates.

**Comment by fesor** [comment]
@radotzki this is stateless service:

```javascript
export default class AboutService {
    constructor(user) {
       this.user = user;
    }

    getName() {
        return `About: ${this.user.getUsername()}`;
    }
}
```

**Pull request closed without merging**